src/lib.rs

```diff
@@ -61,69 +76,253 @@ impl InlinerBuilder {
     ///
     /// Default: `false`.
-    pub fn error_not_found(&mut self, error_not_found: bool) -> &mut Self {
-        self.error_not_found = error_not_found;
+    pub fn track_errors(&mut self, track_errors: bool) -> &mut Self {
+        self.track_errors = track_errors;
         self
     }
 
     /// Parse the source code in `src_file` and return a `syn::File` that has all modules
     /// recursively inlined.
-    pub fn parse_and_inline_modules(&self, src_file: &std::path::Path) -> Result<syn::File, Error> {
+    pub fn parse_and_inline_modules(&self, src_file: &Path) -> Result<syn::File, Error> {
         self.parse_internal(src_file, FsResolver::default())
     }
 
     fn parse_internal<R: FileResolver + Clone>(
         &self,
-        src_file: &std::path::Path,
+        src_file: &Path,
         resolver: R,
     ) -> Result<syn::File, Error> {
-        let mut errors = if self.error_not_found {
+        let mut errors = if self.track_errors {
             Some(vec![])
         } else {
             None
         };
         let result =
             Visitor::<R>::with_resolver(src_file, self.root, errors.as_mut(), Cow::Owned(resolver))
-                .visit();
+                .visit()
+                .map_err(Error::Root)?;
         match errors {
             Some(ref errors) if errors.is_empty() => Ok(result),
             None => Ok(result),
-            Some(errors) => Err(Error::NotFound(errors)),
+            Some(errors) => Err(Error::Inline(IncompleteInlining::new(result, errors))),
         }
     }
 }
 
-/// A source location
-#[derive(Debug, Clone, PartialEq, Eq)]
-pub struct SourceLocation {
-    pub path: PathBuf,
-    pub line: usize,
-    pub column: usize,
-    _private: (),
+/// An error that was encountered while reading, parsing or inlining a module.
+#[derive(Debug)]
+pub enum Error {
+    /// An error happened while reading or parsing the initial file.
+    ///
+    /// For example, the initial file wasn't found or wasn't a valid Rust file.
+    Root(ErrorKind),
+
+    /// The initial file was successfully loaded, but some errors happened while attempting to
+    /// inline modules.
+    Inline(IncompleteInlining),
 }
 
-impl SourceLocation {
-    pub(crate) fn new(path: &Path, span: proc_macro2::Span) -> Self {
-        SourceLocation {
-            path: path.into(),
-            line: span.start().line,
-            column: span.start().column,
-            _private: (),
+impl fmt::Display for Error {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        match self {
+            Error::Root(kind) => write!(f, "error for initial file: {}", kind),
+            Error::Inline(errors) => errors.fmt(f),
         }
     }
 }
 
-/// An error that was encountered while inlining modules
+impl error::Error for Error {}
+
+impl Error {
+    /// Returns true if the error happened while reading or parsing the initial file.
+    pub fn is_initial(&self) -> bool {
```

Comment:
The name of this method should probably be in sync with the name of the variant.

Reply:
The method no longer exists.